### ISSUE TITLE
[a11y]: Revise MultiSelect keyboard selection behavior to not use `enter`

### ISSUE THREAD

**Issue by tay1orjones** [carbon-design-system/carbon#13119]
```[tasklist]
### Tasks
- [ ] The `Enter` key should no longer select items within MultiSelect, only `space` should select items
- [ ] Add tests to cover the behavior change
```

Original guidance from @mbgower in https://github.com/carbon-design-system/carbon/issues/12723:

> ### issue 1 Only Spacebar should be used to select checkboxes in multi-select
> 
> For a single-select dropdown, both Enter and Spacebar can be used to select the item. However, with multi-select it is recommended to only make Spacebar the selection mechanism. There are a number of reasons:
> 1. Spacebar is the traditional mechanism for selection. It is the key that toggles a checkbox, and the multi-select appears as a list of checkboxes.
> 2. It allows Enter to have the function of completing the selections and closing the dropdown. A user _can_ collapse the dropdown by tabbing to the next component, but it is recommended to use a typical 'do' command for this.
> 4. It helps prevent the user from accidentally submitting an incomplete form. There is a slight danger that in a form with a default Submit button, pressing Enter _may_ cause the form to be submitted.

**Comment by shyamtawli** [comment]
Hey @tay1orjones can I try my hands on it ? 
Please assign me.
